src/events/HotKey.hh

```diff
@@ -150,6 +155,7 @@ private:
 	GlobalCommandController& commandController;
 	EventDistributor& eventDistributor;
 	std::optional<Event> lastEvent;
+	std::optional<std::tuple<HotKeyInfo, Event>> postponedEvent;
```

Comment:
I guess a better solution is not to store events, but to keep track of the priority level it is currently in (change `signalEvent()` to include this as a 2nd parameter?).

Reply:
That's possible.

My idea was to have two separate EventListeners (two inner objects) in HotKey. So that we keep the simplicity of exactly one priority per listener object (and so that you don't have to remove the asserts from EventDistributor.cc).

But I've no idea how easy that is. And I don't have much time today to investigate.
I also didn't have time yet to read your full PR in detail. I hope to do that soon (I hope to make time for it this weekend).

Thanks!